File: Lab 4/opt.c
#include <stdio.h>

// #define BITEXTR(a,b) ((a)&1<<((b)-1)?1:0)

int main (void)
{
  unsigned int i = 0;
  puts ("Please input an integer between 0 to 15:");
  scanf ("%u", &i);

  // // this number very 4-bit
  // i &= 0xf;

  printf (
    "%u is %d%d%d%d in binary.\n",
    i,
    // BITEXTR(i, 4), BITEXTR(i, 3), BITEXTR(i, 2), BITEXTR(i, 1)
    (i & 8 ? 1 : 0), (i & 4 ? 1 : 0), (i & 2 ? 1 : 0), (i & 1 ? 1 : 0)
  );
  return 0;
}

File: Lab 4/opt.go
package main

import "fmt"

func main() {
  var i uint = 0
  fmt.Println("Please input an integer between 0 to 15:")
  fmt.Scanf("%d", &i)

  // // this number very 4-bit
  // i &= 0xf;

  fmt.Printf(
    "%d is %d%d%d%d in binary.\n",
    i, bitextr(i, 4), bitextr(i, 3), bitextr(i, 2), bitextr(i, 1))
}

func bitextr(n, bit uint) uint {
  if n & (1 << (bit - 1)) != 0 {
    return 1
  } else {
    return 0
  }
}
